extensions/amp-story/0.1/progress-bar.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {dev} from '../../../src/log';
+import {scale, setStyles} from '../../../src/style';
+import {scopedQuerySelector} from '../../../src/dom';
+import {Services} from '../../../src/services';
+
+
+/**
+ * Progress bar for <amp-story>.
+ */
+export class ProgressBar {
```

Comment:
[`<progress>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress) + CSS not good enough?

Reply:
This is a segmented progress bar, like:

![image](https://user-images.githubusercontent.com/1651960/31123435-ebb7e816-a80d-11e7-9773-80df5b2c429b.png)

Another way to think of this is as a "progress bar of progress bars."  The innermost one could be implemented as the [`<progress>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress) element, but those are difficult to animate with CSS cross-browser, so it may just be worth doing the work to make the `<li>`/`<div>`-based approach work.

Reply:
SGTM.